### DISCUSSION TITLE
Copilot not working on remote

### DISCUSSION THREAD

**Discussion by cossio** [community/community#41704] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I have the following issue in remote servers with Github copilot. It fails to start, with the following error:

![Screenshot 2022-12-14 at 09 51 15](https://user-images.githubusercontent.com/10637482/207549948-44d86596-9274-4627-beaa-b5fa274de973.png)

I have pip installed on the remote. When I click "Show output", this is all that appears:

```
[ERROR] [default] [2022-12-14T08:49:46.824Z] GitHub Copilot could not connect to server. Extension activation failed: "Module 'pip' not installed."
```

**Comment by iamcjwilliams** [comment]
@cossio Can you clarify the remote server? Is copilot working for you on a different server?

**Reply by cossio** [reply]
What do need me to clarify about the remote server? It's a linux box, that I access by SSH on a terminal, or through vscode with the remote extensions.

No, at the moment copilot is not working for me on any server (I typically connect to 3 or 4 linux boxes like this one).

@iamcjwilliams If you need more information, please let me know how I can provide it

**Reply by iamcjwilliams** [reply]
I will have our copilot team investigate this issue and provide an update soon

**Comment by cossio** [comment]
Just want to point out I updated to Copilot v1.65.7705 and still experiencing the issue.

**Reply by iamcjwilliams** [reply]
Please visit this doc to troubleshoot your issue: https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-github-copilot-in-visual-studio-code

**Comment by cossio** [comment]
The output log is just this:

```
[ERROR] [default] [2022-12-20T17:03:22.167Z] GitHub Copilot could not connect to server. Extension activation failed: "Module 'pip' not installed."
```

**Reply by cossio** [reply]
Should I send you the Diagnostics file?

**Comment by osyvokon** [comment]
I have the same error. Remote is WSL2 in my case. Stopped working today with no apparent reason.

**Reply by osyvokon** [reply]
Hm, the error resolved by itself. I did nothing 🤷

**Reply by cossio** [reply]
For me it keeps happening 🙁  Can't use copilot on remotes.

**Comment by kimadeline** [comment]
Hi @cossio,

Do you have other extensions installed in VS Code, both locally and in your remotes? Could you try deactivating Copilot, connecting to your remote, and then re-activating Copilot?

**Reply by cossio** [reply]
Hi, it seems to be resolved now. Maybe there was an update?

**Reply by cossio** [reply]
The issue is happening again. I really don't know what's triggering this.

**Reply by kimadeline** [reply]
Could you answer the questions above, see if these troubleshooting steps might help?

Thanks!

**Reply by cossio** [reply]
Yes I have other extensions. I have deactivated Copilot and re-activated it on the remote, but the issue persists. I also uninstalled Copilot and reinstalled it, but still I have the same issue.

Let me know if I can provide any more info.

**Reply by kimadeline** [reply]
Which other extensions do you have? Could you try disabling all of your extensions except Copilot, and try again? Could you also copy here the contents of the Developer Tools console window (Command Palette > `Developer: Toggle Developer Tools`)?

**Reply by cossio** [reply]
For what it is worth, I'm still seeing this error. Here is the contents of the Developer console.

```
 INFO [attempt 1] Invoking resolveAuthority(ssh-remote)
log.ts:316   ERR [Extension Host] (node:62761) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `Code Helper --trace-deprecation ...` to show where the warning was created)
console.ts:137 [Extension Host] (node:62761) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `Code Helper --trace-deprecation ...` to show where the warning was created)
E @ console.ts:137
log.ts:304  INFO [attempt 1] resolveAuthority(ssh-remote) returned '127.0.0.1:62774' after 43 ms
TMScopeRegistry.ts:47 Overwriting grammar scope name to file mapping for scope source.julia.
Old grammar file: file:///Applications/Visual%20Studio%20Code.app/Contents/Resources/app/extensions/julia/syntaxes/julia.tmLanguage.json.
New grammar file: vscode-remote://ssh-remote%2Blecuona/home/cossio/.vscode-server/extensions/julialang.language-julia-1.38.2/syntaxes/julia_vscode.json
register @ TMScopeRegistry.ts:47
console.ts:137 [Extension Host] Activating extension language-julia
log.ts:316   ERR [Extension Host] (node:550983) DeprecationWarning: uuidv4() is deprecated. Use v4() from the uuid module instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
console.ts:137 [Extension Host] (node:550983) DeprecationWarning: uuidv4() is deprecated. Use v4() from the uuid module instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
E @ console.ts:137
log.ts:304  INFO [perf] Render performance baseline is 14ms
log.ts:316   ERR [Extension Host] [ERROR] [default] [2023-01-30T20:23:10.647Z] GitHub Copilot could not connect to server. Extension activation failed: "Module 'pip' not installed."
console.ts:137 [Extension Host] [ERROR] [default] [2023-01-30T20:23:10.647Z] GitHub Copilot could not connect to server. Extension activation failed: "Module 'pip' not installed."
E @ console.ts:137
console.ts:137 [Extension Host] rejected promise not handled within 1 second: Error: Module 'pip' not installed.
E @ console.ts:137
console.ts:137 [Extension Host] stack trace: Error: Module 'pip' not installed.
    at l.execModule (/home/cossio/.vscode-server/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:7:82791)
    at r.getPackageInformation (/home/cossio/.vscode-server/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:17:42188)
E @ console.ts:137
log.ts:316   ERR Module 'pip' not installed.: Error: Module 'pip' not installed.
    at l.execModule (/home/cossio/.vscode-server/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:7:82791)
    at r.getPackageInformation (/home/cossio/.vscode-server/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:17:42188)
```

**Reply by cossio** [reply]
As for disabling other extensions, note that this issue occurs unpredictably ... unfortunately I cannot have all extensions disabled all the time waiting for this to occur.

**Reply by kimadeline** [reply]
```
log.ts:316   ERR Module 'pip' not installed.: Error: Module 'pip' not installed.
    at l.execModule (/home/cossio/.vscode-server/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:7:82791)
    at r.getPackageInformation (/home/cossio/.vscode-server/extensions/ms-toolsai.jupyter-2022.11.1003412109/out/extension.node.js:17:42188)
```

The error comes from the Jupyter extension, but is being caught and displayed by Copilot somehow. There's an item on our end to not display exceptions coming from other extensions.

I'd recommend filing this issue (pip being reported as not installed) against the [Jupyter extension](https://github.com/microsoft/vscode-jupyter).

**Comment by CarloLucibello** [comment]
I occasionally the same error also when developing locally
